src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/ConnectionTest.php

```diff
@@ -212,26 +212,17 @@ public function testKeepGettingPendingMessages()
             ->willReturn(ResultMockFactory::create(GetQueueUrlResult::class, ['QueueUrl' => 'https://sqs.us-east-2.amazonaws.com/123456789012/MyQueue']));
         $client->expects($this->exactly(2))
             ->method('receiveMessage')
-            ->withConsecutive(
-                [
-                    [
-                        'QueueUrl' => 'https://sqs.us-east-2.amazonaws.com/123456789012/MyQueue',
-                        'MaxNumberOfMessages' => 9,
-                        'WaitTimeSeconds' => 20,
-                        'MessageAttributeNames' => ['All'],
-                        'VisibilityTimeout' => null,
-                    ],
-                ],
-                [
-                    [
-                        'QueueUrl' => 'https://sqs.us-east-2.amazonaws.com/123456789012/MyQueue',
-                        'MaxNumberOfMessages' => 9,
-                        'WaitTimeSeconds' => 20,
-                        'MessageAttributeNames' => ['All'],
-                        'VisibilityTimeout' => null,
-                    ],
-                ]
-            )
+            ->with($this->callback(function (array $config) {
+                $expectedConfig = [
+                    'QueueUrl' => 'https://sqs.us-east-2.amazonaws.com/123456789012/MyQueue',
+                    'MaxNumberOfMessages' => 9,
+                    'WaitTimeSeconds' => 20,
+                    'MessageAttributeNames' => ['All'],
+                    'VisibilityTimeout' => null,
+                ];
+
+                return $expectedConfig == $config;
+            }))
             ->willReturnOnConsecutiveCalls(
```

Comment:
I guess this still needs work

Reply:
Oups, updated this bit. However I'm not 100% sure if the use of `willReturnCallback()` is *the* way to go, so really looking forward for feedbacks on this PR!

Reply:
no idea, let's do the same as we do elsewhere for now